src/NeoSharp.Core/Network/Server.cs

```diff
@@ -76,7 +76,7 @@ public class Server : IServer, IBroadcaster, IDisposable
         #region IServer implementation
 
         /// <inheritdoc />
-        public IReadOnlyCollection<IPeer> ConnectedPeers => _connectedPeers.ToArray();
+        public IReadOnlyCollection<IPeer> ConnectedPeers => _connectedPeers.Where(x => x.IsConnected).ToArray();
```

Comment:
A peer should not be in connected peer collection if it is disconnected.

Reply:
Agree, the problem is this: when the socket is disconnected and is happening, the outside sockets that create the NetworkStream is not been update, therefore, even if we have disconnected peers, the list return all peers as connected.

Maybe a different solution should be in place. For now this fix the bug that we had.

Reply:
We need to notify Server somehow about disconnect to drop from this collection

Reply:
Agree ,, that was not done and the "> Nodes" command was providing wrong information.

This will refactor in the right way when the Reconnect feature will be implemented. I will create a issue to not forget your proposal.

Reply:
Issue #429 created.